src/main/java/io/pravega/connectors/flink/ReaderCheckpointHook.java

```diff
@@ -98,13 +98,17 @@ public String getIdentifier() {
         return checkpointResult;
     }
 
-    @SuppressWarnings("deprecation")
     @Override
     public void restoreCheckpoint(long checkpointId, Checkpoint checkpoint) throws Exception {
         // checkpoint can be null when restoring from a savepoint that
         // did not include any state for that particular reader name
         if (checkpoint != null) {
-            this.readerGroup.resetReadersToCheckpoint(checkpoint);
+            this.readerGroup.resetReaderGroup(ReaderGroupConfig.
+                    builder().
+                    disableAutomaticCheckpoints().
```

Comment:
The `maxOutstandingCheckpointRequest` value will be reset to the default value(3) with change. Do we want to use the same configuration as used in `AbstractStreamingReaderBuilder`?

Reply:
Good to mention. Here we should copy the parameters in the original `readerGroupConfig` instead of create a new one by builder.

Reply:
now the code looks like this, so the newly built readerGroup uses the default parameters

```
.builder()
.maxOutstandingCheckpointRequest(this.readerGroupConfig.getMaxOutstandingCheckpointRequest())
.groupRefreshTimeMillis(this.readerGroupConfig.getGroupRefreshTimeMillis())
.disableAutomaticCheckpoints()
.startFromCheckpoint(checkpoint)
.build()
```